share/normalize/mods.py

```diff
@@ -0,0 +1,431 @@
+    )
+
+    # may need to check for something else if displayForm isn't there
+    name = tools.Try(tools.RunPython(force_text, ctx['mods:displayForm']))
+
+    related_agents = tools.Map(tools.Delegate(IsAffiliatedWith), tools.Concat(tools.Try(
+        tools.Filter(lambda x: bool(x), tools.RunPython(force_text, ctx['mods:affiliation']))
+    )))
+
+    class Extra:
+        name_type = tools.Try(ctx['@type'])
+        name_part = tools.Try(ctx['mods:namePart'])
+        affiliation = tools.Try(ctx['mods:affiliation'])
+        description = tools.Try(ctx['mods:description'])
+        display_form = tools.Try(ctx['mods:displayForm'])
+        etal = tools.Try(ctx['mods:etal'])
+        name_identifier = tools.Try(ctx['mods:nameIdentifier'])
+
+
+class MODSAgentIdentifier(Parser):
+    schema = 'AgentIdentifier'
+
+    uri = ctx
+
+
+class MODSWorkIdentifier(Parser):
+    schema = 'WorkIdentifier'
+
+    uri = ctx
+
+
+class MODSSubject(Parser):
+    schema = 'Subject'
+
+    name = ctx
+
+
+class MODSThroughSubjects(Parser):
+    schema = 'ThroughSubjects'
+
+    subject = tools.Delegate(MODSSubject, ctx)
+
+
+class MODSTag(Parser):
+    schema = 'Tag'
+
+    name = ctx
+
+
+class MODSThroughTags(Parser):
+    schema = 'ThroughTags'
+
+    tag = tools.Delegate(MODSTag, ctx)
+
+
+class MODSRelatedWork(Parser):
+    schema = 'CreativeWork'
+
+    identifiers = tools.Map(tools.Delegate(MODSWorkIdentifier), ctx)
+
+    class Extra:
+        identifier = ctx
+
+
+class MODSWorkRelation(Parser):
+    schema = 'WorkRelation'
+
+    def get_creative_work(self):
+        return MODSCreativeWork
+
+    related = tools.Delegate(get_creative_work, ctx)
+
+
+class MODSAgentWorkRelation(Parser):
+    schema = 'AgentWorkRelation'
+
+    agent = tools.Delegate(MODSAgent, ctx)
+    cited_as = tools.RunPython(force_text, tools.Try(ctx['mods:displayForm']))
+
+
+class MODSContributor(MODSAgentWorkRelation):
+    schema = 'Contributor'
+
+
+class MODSCreator(MODSContributor):
+    schema = 'Creator'
+
+    order_cited = ctx('index')
+
+
+class MODSPublisher(Parser):
+    schema = 'Publisher'
+
+    agent = tools.Delegate(MODSAgent.using(schema=tools.GuessAgentType(ctx, default='organization')), ctx)
+
+
+class MODSCreativeWork(Parser):
+    default_type = 'CreativeWork'
+    type_map = None
+
+    schema = tools.RunPython(
+        'get_schema',
+        tools.OneOf(
+            tools.RunPython(force_text, ctx['mods:genre']),
+            tools.Static(None)
+        )
+    )
+
+    title = tools.Join(tools.RunPython(force_text, tools.Try(ctx['mods:titleInfo']['mods:title'])))
+
+    # TODO
+    # This attribute is used with shareable="no" for data that may be proprietary or is rights
+    # protected and should not be used outside of a local system (such as providing to harvesters).
+    description = tools.Join(tools.RunPython(force_text, tools.Try(ctx['mods:abstract'])))
+
+    # do we want isbns?
+    identifiers = tools.Map(
```

Comment:
Bit of an odd one, yes if its in a book but not a book section.

Probably fine to collect for the moment.